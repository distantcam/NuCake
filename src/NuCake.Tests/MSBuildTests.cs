using System.Runtime.CompilerServices;
using ApprovalTests;
using ApprovalTests.Reporters;
using Xunit;

[UseReporter(typeof(DiffReporter))]
public class MSBuildTests
{
    [MethodImpl(MethodImplOptions.NoInlining)]
    [Fact]
    public void NoAttributesGivesWarnings()
    {
        Approvals.Verify(Tools.RunMSBuild(@"..\..\..\SampleProject\SampleProject.csproj"));
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    [Fact]
    public void SettingTitleRemovesWarning()
    {
        Approvals.Verify(Tools.RunMSBuild(@"..\..\..\SampleProject\SampleProject.csproj", "SetTitle"));
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    [Fact]
    public void SettingDescriptionRemovesWarning()
    {
        Approvals.Verify(Tools.RunMSBuild(@"..\..\..\SampleProject\SampleProject.csproj", "SetDescription"));
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    [Fact]
    public void SettingCompanyRemovesWarning()
    {
        Approvals.Verify(Tools.RunMSBuild(@"..\..\..\SampleProject\SampleProject.csproj", "SetCompany"));
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    [Fact]
    public void SettingEverythingRemovesAllWarning()
    {
        Approvals.Verify(Tools.RunMSBuild(@"..\..\..\SampleProject\SampleProject.csproj", "SetTitle", "SetDescription", "SetCompany", "SetInformationalVersion"));
    }
}