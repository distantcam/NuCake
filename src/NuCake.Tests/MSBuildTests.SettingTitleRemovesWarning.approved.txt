__________________________________________________
Project "[PROJECT_DIRECTORY]\SampleProject.csproj" (Rebuild target(s)):
Target CoreClean:
    Deleting file "[PROJECT_DIRECTORY]\bin\[CONFIG]\SampleProject.dll".
    Deleting file "[PROJECT_DIRECTORY]\bin\[CONFIG]\SampleProject.pdb".
    Deleting file "[PROJECT_DIRECTORY]\obj\[CONFIG]\SampleProject.dll".
    Deleting file "[PROJECT_DIRECTORY]\obj\[CONFIG]\SampleProject.pdb".
Target GenerateTargetFrameworkMonikerAttribute:
  Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
Target CoreCompile:
    [DOTNET_FRAMEWORK]\Csc.exe /noconfig /nowarn:1701,1702 /nostdlib+ /platform:AnyCPU /errorreport:prompt /warn:4 /define:SetTitle /highentropyva- /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\mscorlib.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\System.Core.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\System.dll" /debug:pdbonly /filealign:512 /optimize+ /out:obj\[CONFIG]\SampleProject.dll /target:library /utf8output AssemblyInfo.cs Folder\ClassInAFolder.cs MyClass.cs "[TEMP_DIRECTORY].NETFramework,Version=v4.0.AssemblyAttributes.cs"
Target CopyFilesToOutputDirectory:
    Copying file from "obj\[CONFIG]\SampleProject.dll" to "bin\[CONFIG]\SampleProject.dll".
    SampleProject -> [PROJECT_DIRECTORY]\bin\[CONFIG]\SampleProject.dll
    Copying file from "obj\[CONFIG]\SampleProject.pdb" to "bin\[CONFIG]\SampleProject.pdb".
Target CreateNuGetPackage:
    [PROJECT_DIRECTORY]\NuCake.targets(7,5): warning : No company name found. Add a AssemblyCompanyAttribute to your project.
    Package created -> [PROJECT_DIRECTORY]\NuGetBuild\SampleProject 1.0.0.nupkg
    Symbols created -> [PROJECT_DIRECTORY]\NuGetBuild\SampleProject 1.0.0.symbols.nupkg
Done building target "CreateNuGetPackage" in project "SampleProject.csproj".
Done building project "SampleProject.csproj".
Build succeeded.
[PROJECT_DIRECTORY]\NuCake.targets(7,5): warning : No company name found. Add a AssemblyCompanyAttribute to your project.
    1 Warning(s)
    0 Error(s)